python/parquet_tools/csv2pq_schema.py

```diff
@@ -0,0 +1,260 @@
+
+
+# *****************************************************************************
+# *           T y p e   C o n v e r s i o n   I n f o r m a t i o n           *
+# *****************************************************************************
+
+class TypeInfo:
+
+    table = {'int':     np.int32,   'short':    np.int16, 'long':  np.int64,
+             'bigint':  np.int64,   'smallint': np.int16,
+             'int8':    np.int8,    'int32':    np.int32, 'int64': np.int64,
+             'float':   np.float32, 'double':   np.float64,
+             'float32': np.float32, 'float64':  np.float64,
+             'char':    np.bytes_,  'bool':     np.bool_
+             }
+
+    isInt = {'int':     True,       'short':    True,     'long':  True,
+             'bigint':  True,       'smallint': True,
+             'int8':    True,       'int32':    True,     'int64': True
+             }
+
+
+# *****************************************************************************
+# *                                a d d C o l                                *
+# *****************************************************************************
+
+def addCol(cName):
+    "Add a column to the column list and record size of longest name."
+
+    CmdInfo.colNames.append(cName)
+    if len(cName) > CmdInfo.cNameMax: CmdInfo.cNameMax = len(cName)
+
+
+# *****************************************************************************
+# *                               g e t D e c T                               *
+# *****************************************************************************
+
+def getDecT(dArgs):
+    "Convert decimal type to appropriate numpy type."
+
+    # Find the closing paren a strip off the remianing text
+    #
+    n = dArgs.find(')')
+    if n < 1: return ''
+    xArgs = dArgs[0:n]
+
+    # Split this into one or two args and convert
+    #
+    pSpec = xArgs.split(',')
+    try:
+        pVal1 = int(pSpec[0])
+    except Exception:  # We don't care what he exception is
+        return ''
+    if len(pSpec) == 1:
+        pVal2 = 0
+    else:
+        try:
+            pVal2 = int(pSpec[1])
+        except Exception:  # We don't care what gthe exception is
+            return ''
+
+    # Return the an int type if this has no digits after the decimal point
+    #
+    if pVal2 == 0:
+        if pVal1 < 3:
+            return 'int8'
+        elif pVal1 < 5:
+            return 'int16'
+        elif pVal1 < 10:
+            return 'int32'
+        else:
+            return 'int64'
+
+    # Return appropriate float
+    #
+    if pVal1 < 7: return 'float32'
+    return 'float64'
+
+
+# *****************************************************************************
+# *                               g e t T y p e                               *
+# *****************************************************************************
+
+def getType(cName, cType, cNull, colNulls):
+    "Record correct type for a particular column."
```

Comment:
Well, that seems to be controversial. Full docstrings for public API's  are certainly warranted but a some people think that is going way to far for private API's. It's totally unclear where LSST stands on that controversy.

Reply:
I prefer proper docstrings for all functions wherever possible. Mainly because you don't know who is going to be working on this code in the future and it helps future developers work out what is going on when they don't have to guess the expected parameter types.  Using modern python type hinting is certainly a possibility.

Reply:
I don't think that “it's totally unclear where LSST stands”, but if you can suggest improvements to the wording of https://developer.lsst.io/python/style.html#docstrings-should-be-written-for-all-public-modules-functions-classes-and-methods to help make it clearer, that would be great!